### ISSUE TITLE
epic: removal of the legacy helm charts

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#20441]
* [x] Blocker: https://github.com/istio/istio/issues/20336
* [x] Blocker: https://github.com/istio/istio/issues/20335
* [x] Blocker: https://github.com/istio/istio/issues/20334
* [x] Blocker: need to port the pkg/inject/ tests that depend on these new charts https://github.com/istio/istio/pull/20469
* [x] Cleanup: remove support for helm from test/integration/
* [x] Cleanup: removal all of the legacy e2e test framework https://github.com/istio/istio/pull/21896
* [x] Cleanup: remove the sidecar injector main.go, dockerfile, etc
* [x] Cleanup: remove the citadel main.go, dockerfile, etc https://github.com/istio/istio/pull/21837
* [x] Remove all documtation after 1.5 branches that references helm. https://github.com/istio/istio.io/pull/6458
* [ ] Cleanup: remove the galley main.go, dockerfile, etc

**Comment by howardjohn** [comment]
https://github.com/istio/istio/pull/21185 does the core of this, the rest is cleanup

**Comment by tmshort** [comment]
Is there something that explains why the helm charts are being removed? Will istioctl be the only installation mechanism? Or will it be istio/intaller or is that moving to istio/manifests? Looking for a user's (not developer's) perspective.

**Comment by howardjohn** [comment]
@tmshort istioctl uses helm charts under the hood the live in the manifests/ folder (this was the istio/installer repo, we merged the two repos). These charts can be installed using Helm or istioctl. 

Maintaining two different sets of helm charts was not feasible or productive for the project, so we have decided to remove the legacy ones. Helm can still be used with the new charts.

By the time we ship 1.6 we will have a new upgrade method (https://docs.google.com/document/d/1d1z5PC8wvh9QiR3NesAaPm6w2HnE4-KIfjB7GB-_Wf8/edit#) that is much safer so migration from legacy charts to the new ones will be safe
